ir/src/tests/mod.rs

```diff
@@ -379,6 +379,167 @@ fn transition_constraints_using_parens() {
     assert!(result.is_ok());
 }
 
+#[test]
+fn random_values_fixed_list() {
+    let source = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [16]
+    integrity_constraints:
+        enf c' = $rand[3] + 1
+    boundary_constraints:
+        enf c.first = $rand[10] * 2
+        enf c.last = 1";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_ok());
+}
```

Comment:
nit: here and on all tests, I'd prefer to put the `boundary_constraints` before the `integrity_constraints`, since that's generally what we do, and consistency tends to be helpful